src/strategies/staked-defi-balance/index.ts

```
@@ -15,40 +19,80 @@ export async function strategy(
+        break;
+      case 'standard':
+        abi = standardStakingAbi[0] as ABI;
+        break;
+      default:
+        throw new Error(`Invalid stakingType: ${params.stakingType}`);
+    }
 
-  const addressScores = Object.fromEntries(
-    addresses.map((address) => {
-      const normalizedAddress = getAddress(address);
-      const stakedBalance = stakesMapped[normalizedAddress];
-      const formattedStakedBalance = parseFloat(
-        formatUnits(stakedBalance, options.decimals)
+
+    const stakingCalls = addresses.map((address) => {
+      const inputs = abi.inputs.map((input) => {
+        if (input.name === 'id') {
+          return params.tokenContractAddress;
+        } else if (input.name === 'staker' || input.name === 'account') {
+          return address;
+        }
+      });
+      return [stakingPoolContractAddress, abi.name, inputs];
+    });
+
+    try {
+      const stakes = await multicall(
+        paramNetwork,
+        getProvider(paramNetwork),
+        [abi],
+        stakingCalls,
+        { blockTag: paramSnapshot }
       );
-      return [
-        normalizedAddress,
-        stakedBalance.gte(options.minStakedBalance) ? formattedStakedBalance : 0
-      ];
-    })
+
+      const stakesMapped = {};
+      for (let i = 0; i < addresses.length; i++) {
+        stakesMapped[getAddress(addresses[i])] = stakes[i][0];
+      }
+
+      addresses.forEach((address) => {
+        const normalizedAddress = getAddress(address);
+        const stakedBalance = stakesMapped[normalizedAddress];
+        const formattedStakedBalance = parseFloat(
+          formatUnits(stakedBalance, params.decimals)
+        );
+
+        // Initialize address score if it doesn't exist
+        if (!addressScores[normalizedAddress]) {
+          addressScores[normalizedAddress] = 0;
+        }
+
+        addressScores[normalizedAddress] += formattedStakedBalance;
+      });
+    } catch (error) {
+      console.error('Error in multicall:', error);
+    }
```

If there is some network error, the voting power can change and can cause a lot of issues. better to throw an error if there is an error :)

---

This is also resolved